spec/services/export/report_spec.rb

```diff
@@ -29,26 +31,31 @@
     end
 
     describe "#rows" do
-      it "returns the rows" do
-        rows = subject.rows
-        expect(rows.count).to eq @activities.count
-        expect(rows.fetch(@activities.first.id)).to include(@activities.first.roda_identifier)
-        expect(rows.fetch(@activities.last.id)).to include(@activities.last.roda_identifier)
+      it "returns the rows ordered by level" do
+        rows = subject.rows.to_a
+
+        expect(rows.count).to eq 2
+        expect(rows.first[1]).to include(@activity.roda_identifier)
+        expect(rows.last[1]).to include(@third_party_project.roda_identifier)
```

Comment:
Hmm, I am not sure I fully agree? If I leave the spec as is and change the implementation to:

```ruby
.order(level: :desc) 
```

The roda identrifiers come our the wrong way and the spec fails, because we load `rows.first` and `rows.last`.

I do agree that it is not super obvious to read the spec though!

Reply:
@edavey It took me a while to get to where you are!

I've got this now:

```ruby
      describe "ordering" do
        it "returns the rows ordered descending by level" do
          relation =
            double(
              ActiveRecord::Relation,
              order: Activity.all,
              pluck: [@project.id, @third_party_project.id]
            )
          finder_double = double(Activity::ProjectsForReportFinder, call: relation)
          allow(Activity::ProjectsForReportFinder).to receive(:new).and_return(finder_double)

          subject.rows

          expect(relation).to have_received(:order).with(level: :desc)
        end
      end
```

I've pushed it as a seperate refactor commit on the end.